File: DailyChallenges/CPP/LeetCode_443.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int cnt = 0;
        string s = "";
        for (auto ch: chars){
            if (ch == *s.rbegin()) ++cnt;
            else{
                if (cnt > 1) s = s + to_string(cnt);
                cnt = 1;
                s += ch;
            }
        }
        if (cnt>1) s += to_string(cnt);
        for (int i=0; i< s.length(); ++i)
            chars[i] = s[i];
        return s.length();
    }
};

File: DailyChallenges/Scala/LeetCode_443.scala
object Solution {
    def compress(chars: Array[Char]): Int = {
        var j = 0; var cnt = 0;
        for (i <- 0 until chars.length){
            if (chars(i) == chars(j)) cnt += 1
            else{
                j += 1
                if (cnt > 1){
                    for (ch <- cnt.toString){
                        chars(j) = ch
                        j += 1
                    }
                }
                chars(j) = chars(i)
                cnt = 1
            }
        }
        if (cnt > 1){
            for (ch <- cnt.toString){
                j += 1
                chars(j) = ch
            }
        }
        j+1
    }
}
